### DISCUSSION TITLE
LAB 05.01 Task 3: ML Metric vs Business Metric

### DISCUSSION THREAD

**Discussion by Bexzho** [rramosp/ai4eng.v1#64] [05 - GESTIÓN DE MODELOS]
Hola.
Para solucionar la tarea número 3 cree un código que al usarlo con el ejemplo que se da funciona correctamente, este es el código:

<img width="371" alt="2" src="https://user-images.githubusercontent.com/58709463/135349645-d72f3bed-41f5-4cb1-9e72-1afe2b840005.png">

El problema es que al usarlo para desarrollar el ejercicio y verificar me dice que está incorrecto. Para crear la matriz de predicciones donde cada fila tiene el 10%, 20% , etc. de accuracy, lo que hice fue usar el vector de predicciones perfectas y, por ejemplo, para hallar el vector de predicciones con un 10% de accuracy lo que hice fue multiplicar 20 por 0.1 lo que es igual a 2 y por tanto dejo las dos primeras entradas del vector de predicciones perfectas y el resto de las entradas las pongo incorrectas (inversas a las entradas del vector de predicciones perfectas) y así con las demás.

Estos son el vector de matrices y el vector de PNL que sale del código mostrado anteriormente:

<img width="576" alt="1" src="https://user-images.githubusercontent.com/58709463/135350936-0cff8352-2dbc-4ce7-890d-4c037c64a4ee.png">

Además, el vector de predicciones perfectas es:

![image](https://user-images.githubusercontent.com/58709463/135351719-80a7e2ab-a4c4-4d77-b466-f3013a419b49.png)

No tengo idea por que ocurre esto, cualquier ayuda se las agradezco.

**Comment by rramosp** [comment]
felipe, nos puedes ayudar con esta duda?

gracias!!

El mié, 29 sept 2021 a las 16:29, Alexander R.G ***@***.***>)
escribió:

> Hola.
> Para solucionar la tarea número 3 cree un código que al usarlo con el
> ejemplo que se da funciona correctamente, este es el código:
>
> [image: 2]
> <https://user-images.githubusercontent.com/58709463/135349645-d72f3bed-41f5-4cb1-9e72-1afe2b840005.png>
>
> El problema es que al usarlo para desarrollar el ejercicio y verificar me
> dice que está incorrecto. Para crear la matriz de predicciones donde cada
> fila tiene el 10%, 20% , etc. de accuracy, lo que hice fue usar el vector
> de predicciones perfectas y, por ejemplo, para hallar el vector de
> predicciones con un 10% de accuracy lo que hice fue multiplicar 20 por 0.1
> lo que es igual a 2 y por tanto dejo las dos primeras entradas del vector
> de predicciones perfectas y el resto de las entradas las pongo incorrectas
> (inversas a las entradas del vector de predicciones perfectas) y así con
> las demás.
>
> Estos son el vector de matrices y el vector de PNL que sale del código
> mostrado anteriormente:
>
> [image: 1]
> <https://user-images.githubusercontent.com/58709463/135350936-0cff8352-2dbc-4ce7-890d-4c037c64a4ee.png>
>
> No tengo idea por que ocurre esto, cualquier ayuda se las agradezco.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/64>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIPR3ZG5N7PTEI2FYIDUEOAMTANCNFSM5FA3DFZA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Reply by carlosdac55** [reply]
Yo tengo exactamente el mismo problema, el array de 9x20 con las predicciones de acuerdo al porcentaje que le corresponde a cada fila esta bien, sin embargo, al enviar el error que me devuelve es que los valores de PNL no son los correctos, lo he intentado con los valores de PNL como un array y como una lista con dos decimales y en ambas ocasiones tengo el mismo error, al usar los valores del ejemplo obtengo los mismos resultados. Entonces no se que hacer.

**Comment by camilo1109** [comment]
buenas tardes, 

respecto a este laboratorio tengo una duda con la tercera actividad.

La variable pset es la matriz, donde cada fila tiene cierto accuracy en las predicciones, pero no logro entender que es lo que hay que poner en el vector de 9 posiciones llamado pnl, ya que en primer punto ese pnl era un array cada señal y ahora parece que debe ser un numero para cada señal 

Muchas gracias por la explicación.
